vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -594,6 +611,16 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 		d.Set("power_state", "suspended")
 	}
 
+	// Get USB Controller information
+	var isUSBPresent bool
+	for _, dev := range vprops.Config.Hardware.Device {
+		if _, ok := dev.(*types.VirtualUSBController); ok {
+			isUSBPresent = true
+			break
+		}
+	}
+	_ = d.Set("usb_controller", isUSBPresent)
```

Comment:
Maybe something that would return more like this since more than one controller can be present but only one per version.

```json
[
    {
        "usb_controller": true,
        "usb_version": "2.0"
    },
    {
        "usb_controller": false,
        "usb_version": "3.1"
    }
]
```

Reply:
Maybe something like this?

```go
// Initialize the presence map using usbControllerVersions
usbPresence := make(map[string]bool)
for _, version := range usbControllerVersions {
    usbPresence[version] = false
}

// Get USB Controller information
for _, dev := range vprops.Config.Hardware.Device {
    switch dev.(type) {
    case *types.VirtualUSBController:
        usbPresence["2.0"] = true
    case *types.VirtualUSBXHCIController:
        usbPresence["3.1"] = true
    }
}

// Convert the map to a slice of maps for setting the attribute
var usbControllers []map[string]interface{}
for version, present := range usbPresence {
    usbControllers = append(usbControllers, map[string]interface{}{
        "usb_controller": present,
        "usb_version":    version,
    })
}

_ = d.Set("usb_controller", usbControllers)
```

Reply:
if i can get a lab up i can test this, just not sure i want to chage it until i can test